cmd/generate_repo_config/generate_repo_config.go

```diff
@@ -104,8 +111,12 @@ func generateRepoConfig(configDest, configSource string) ([]string, error) {
 	for _, r := range sortedFiles {
 		for _, d := range r.Directives {
 			// skip repository_macro directives, because for the repo config we flatten
-			// macros into one file
-			if d.Key != "repository_macro" {
+			// macros into one file. skip repository directives because they are merged
+			// into repos already.
+			switch d.Key {
+			case _repositoryMacroKey, _repositoryKey:
+				continue
+			default:
```

Comment:
Right now this case will never be hit because `repositoryMacroKey` and `repositoryKey` are the only workspace directives. Maybe we should assume that `ListRepositories` will process all workspace directives so there is no need to add them into the merged file? That way we can remove all of this logic.